### ISSUE TITLE
[a11y]: Add regions to PageHeaders

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#8048]
### Package

@carbon/ibm-products

### Package version

v2.72.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

Header is not defined with in a labelled region. Screen reader is not announcing any region for the header section. Screen reader users feel difficulty to identify the content boundaries of the mentioned sections, also affect quick navigation  between the content sections.

**Recommended fix**
To resolve this issue, explicitly define the role of the header section using the appropriate ARIA landmark role (`role="banner"` or update `section` to `header`). This ensures that assistive technologies can properly identify and navigate to the header content.

**User impact**
The absence of a defined landmark roll along with appropriate label for the content sections within the page  affects users who rely on assistive technologies, such as screen readers, to navigate and understand the structure of a web page. Without a defined role, these users may encounter difficulty in quickly locating and accessing important header information, leading to a less efficient and potentially frustrating browsing experience.

### Acceptance criteria
- [ ] Add `role="banner"` or update `section`/`div` to use `header` in page headers
  - [ ] PageHeader v1
  - [ ] PageHeader v2
  - [ ] PageHeader WC

### WCAG 2.1 Violation

1.3.1. Info & Relationships

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/overview-welcome--overview

### Steps to reproduce

1. Navigate to simple page header or new page header default
2. Use a screen reader to navigate to the page title

No region is read out to the user.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
